### DISCUSSION TITLE
Panel Layout

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#43] [General]
I noticed that one can actually move graphics around on the panel. Great. However it is disappointing to be unable to finely nudge elements into their place using the arrow keys. It would help enormously to be able to do that.

Secondly the frame that appears eg around the keyboard is a bit of a nuisance as it distracts one's attention from the actual object and renders it that bit more difficult to position correctly using visually cues. 

![image](https://github.com/larspalo/GOODF/assets/6999990/daeeb5b0-f541-4c29-b306-913e1c373fac)

The appearance of pixel positions x and y are helpful but appear rather small and faint on the screen. They will be most effective when it becomes possible to nudge things more accurately using arrow keys.... one pixel at a time.

**Comment by larspalo** [comment]
> to finely nudge elements into their place using the arrow keys

Yes, I'll eventually add that key (arrows) handling too.

> the frame that appears eg around the keyboard

The reason the outline doesn't match the actual manual bitmaps is likely because the values in the displaymetrics doesn't match the ones actually used by the manual keys. Try reverting them back (in the displaymetrics) to default values of manual key width 12 and height 32. The pedal should be width 7 and height 40. For easy handling I've just used the provided default values that GO uses for default layout calculations and not actually bothered calculated the used bitmaps combined widths and actual height. So, for now, try matching the displaymetrics values to what is actually used and see if that makes any difference. (Also the amount of keys actually on the manual could make difference if only some of them are displayed - but I'm not sure about that)

**Reply by Bumblebee001** [reply]
I never touched those parameters! Setting them as you say sorted the issue.... 
So I assume you set them at those values inadvertently and they will be corrected in an update.
